src/components/study-pane.js

```diff
@@ -771,6 +786,88 @@ const StudyPane = (props) => {
         network.useEquipment(equipments.lines);
     }, [network]);
 
+    function showLineMenu(line, x, y) {
+        setLineMenu({
+            position: [x, y],
+            lineToApply: line,
+            display: true,
+        });
+    }
+
+    function closeLineMenu() {
+        setLineMenu({
+            display: false,
+        });
+    }
+
+    function handleLockout(lineId) {
+        lockoutLine(studyUuid, lineId)
+            .then((response) => {
+                if (response.status === 304) {
+                    enqueueSnackbar(
+                        intl.formatMessage({ id: 'UnableToLockoutLine' }),
+                        {
+                            variant: 'info',
+                            persist: false,
+                            style: { whiteSpace: 'pre-line' },
+                        }
+                    );
+                }
+            })
+            .then(closeLineMenu);
+    }
+
+    function handleTrip(lineId) {
```

Comment:
Move function to line-menu.js ?

Reply:
Keeping function outside of the component file would allow to reuse the component with different behaviour

Reply:
Done in https://github.com/gridsuite/gridstudy-app/pull/332/files